packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp

```diff
@@ -0,0 +1,248 @@
+}
+
+@scenario
+@scenarioDoc("""
+
+Expected input body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1,
+  "innerModel": {
+    "name": "InnerMadge",
+    "description": "innerDesc"
+  },
+  "intArray": [1, 2, 3]
+}
+```
+
+Expected response body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1,
+  "innerModel": {
+    "name": "InnerMadge",
+    "description": "innerDesc"
+  },
+  "intArray": [1, 2, 3]
+}
+```
+""")
+@doc("Test content-type: application/merge-patch+json with required body")
+@route("/create/resource")
+@put
+op createResource(@body body: Resource): Resource;
+
+@scenario
+@scenarioDoc("""
+Should serialize null values with merge-patch+json enabled.
+
+Expected input body:
+```json
+{
+  "description": null,
+  "map": {
+    "key": {
+      "description": null
+    },
+    "key2": null
+  },
+  "array": null,
+  "intValue": null,
+  "floatValue": null,
+  "innerModel": null,
+  "intArray": null
+}
+```
+
+Expected response body:
+```json
+{
+  name: "Madge",
+  map: {
+    key: {
+      name: "InnerMadge"
+    }
+  }
+}
+```
+""")
+@doc("Test content-type: application/merge-patch+json with required body")
+@route("/update/resource")
+@patch
+op updateResource(
+  @header("content-type") contentType: "application/merge-patch+json",
+  @body body: ResourcePatch,
+): Resource;
+
+@scenario
+@scenarioDoc("""
+
+Expected input body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1,
+  "innerModel": {
+    "name": "InnerMadge",
+    "description": "innerDesc"
+  },
+  "intArray": [1, 2, 3]
+}
+```
+
+Expected response body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1,
+  "innerModel": {
+    "name": "InnerMadge",
+    "description": "innerDesc"
+  },
+  "intArray": [1, 2, 3]
+}
+```
+""")
+@doc("Test content-type: application/merge-patch+json with optional body")
+@route("/create/resource/optional")
+@put
+op createOptionalResource(
+  @header("content-type") contentType: "application/merge-patch+json",
+  @body body?: Resource,
+): Resource;
```

Comment:
Do we need this test?

`@put` may be useful for shared model. `body?` is useful for json-merge-patch. But `@put op create(body?)` seems not necessary?

Reply:
I just add it to align with the required case (create+update).  I can remove it.

Reply:
Removed 'create optional resource' case